File: src/parser_utils/ast_loc_utils.ml
(*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *)

(* Provides utilities for converting AST locations between ALoc.t and Loc.t *)

let loc_to_aloc_mapper : (Loc.t, Loc.t, ALoc.t, ALoc.t) Flow_polymorphic_ast_mapper.mapper =
  object
    inherit [Loc.t, Loc.t, ALoc.t, ALoc.t] Flow_polymorphic_ast_mapper.mapper

    method on_loc_annot = ALoc.of_loc

    method on_type_annot = ALoc.of_loc
  end

File: rust_port/crates/flow_parser_utils/src/ast_loc_utils.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

use dupe::Dupe;
use flow_aloc::ALoc;
use flow_parser::loc::Loc;
use flow_parser::polymorphic_ast_mapper::LocMapper;

pub struct LocToALocMapper;

impl LocMapper<Loc, Loc, ALoc, ALoc> for LocToALocMapper {
    fn on_loc_annot(&mut self, loc: &Loc) -> Result<ALoc, !> {
        Ok(ALoc::of_loc(loc.dupe()))
    }

    fn on_type_annot(&mut self, annot: &Loc) -> Result<ALoc, !> {
        Ok(ALoc::of_loc(annot.dupe()))
    }
}
